tests/unit/gtBSMetadata.cpp

```diff
@@ -0,0 +1,169 @@
+		\"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";\
+	result = read_gbs_metadata(bitstream_guid_invalid3, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// valid input bitstream
+	uint8_t bitstream_guid_invalid4[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640, \"afu-image\":\
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156, \
+		\"power\": 50, \"interface-uuid\": \"1a422218-6dba-448e-b302-425cbcde1406\", \
+		\"magic-no\": 488605312, \"accelerator-clusters\": [{\"total-contexts\":1,\
+		\"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";
+
+	result = read_gbs_metadata(bitstream_guid_invalid4, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid_invalid5[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640 }, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid5, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid_invalid6[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640, \"afu-image\": \
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156,\
+		\"power\": 50,  \"magic-no\": 488605312, \"accelerator-clusters\": \
+		[{\"total-contexts\": 1, \"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid6, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid_invalid7[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640, \"afu-image\": \
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156, \
+		\"power\": 50, \"interface-uuid\": \"1a422218-6dba-448e-b302-425cbcde1406\", \
+		\"magic-no\": 488605312, \"accelerator-clusters\":\
+		[{\"total-contexts\": 1, \"name\": \"nlb_400\"}]}, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid7, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid_invalid8[] = "XeonFPGA·GBSv001\00\00\00\00{\"version\": 640, \"afu-image\":\
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156, \
+		\"power\": 50, \"interface-uuid\": \"1a422218-6dba-448e-b302-425cbcde1406\", \
+		\"magic-no\": 488605312, \"accelerator-clusters\": [{\"total-contexts\":1,\
+		\"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid8, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+}
+
+/**
+* @test    bs_metadata_02
+* @brief   Tests: validate_bitstream_metadata
+* @details validate_bitstream_metadata verifies metadata
+*          Then the return value is FPGA_OK<br>
+*/
+TEST(LibopaecBSMetadataCommonMOCKHW, bs_metadata_02) {

```

Comment:
The comments for this test don't seem to match up with what the test is doing.
Also, how different is this test from the others above?